osquery/tables/system/darwin/user_groups.mm

```diff
@@ -51,38 +51,47 @@ void genODEntries(ODRecordType type, std::set<std::string>& names) {
   }
 }
 
+void setGroupRow(Row& r, group* grp) {
+  r["groupname"] = TEXT(grp->gr_name);
+  r["gid"] = BIGINT(grp->gr_gid);
+  r["gid_signed"] = BIGINT((int32_t)grp->gr_gid);
+}
+
 QueryData genGroups(QueryContext& context) {
   QueryData results;
   if (context.constraints["gid"].exists(EQUALS)) {
     auto gids = context.constraints["gid"].getAll<long long>(EQUALS);
     for (const auto& gid : gids) {
-      Row r;
       struct group* grp = getgrgid(gid);
-      r["gid"] = BIGINT(gid);
-      if (grp != nullptr) {
-        r["groupname"] = std::string(grp->gr_name);
-        r["gid_signed"] = BIGINT((int32_t)grp->gr_gid);
+      if (grp == nullptr) {
+        continue;
       }
+      Row r;
+      setGroupRow(r, grp);
       results.push_back(r);
     }
   } else {
     std::set<std::string> groupnames;
     genODEntries(kODRecordTypeGroups, groupnames);
     for (const auto& groupname : groupnames) {
+      // There may be cases where genODEntries produces responses that
```

Comment:
This is interesting, where is the documentation for this?

Reply:
I have not dug into docs. But the code paths look very different, and so I assume there may be data discrepancies. I tend to favor the defensive stance here of assuming something will mis-align. 

I don't understand what opendirectory queries for `/Local/Default` and under what, if any, conditions it would query a remote server. In contrast the docs for `getpwnam` are that it `operate[s] on the password database file`. Though there is a call out for remote password servers.

In conclusion ¯\_(ツ)_/¯ I'm being skittish around APIs I'm not deeply familiar with